hr_attendance_reason/controllers/main.py

```diff
@@ -11,7 +12,23 @@ class HrAttendance(HrAttendance):
     @route("/hr_attendance/attendance_user_data", type="json", auth="user")
     def user_attendance_data(self):
         response = super().user_attendance_data()
-        response.update(self._get_attendance_reason_settings(request.env.company))
+        # try to get the company of the employee to show the correct reasons from
+        # the employees company
+        company = (
+            request.env.user.employee_id.company_id
+            if request.env.user.employee_id
+            else request.env.company
+        )
```

Comment:
Why is this change necessary? IMO I think it is correct as it was before (`request.env.company`)

Reply:
The reason is that Odoo book the attendance not in the company which is currently selected but in the company where the employee of the user is configured. That means that it is necessary that also the reasons come from the company of the employee because otherwise the company of the attendance and the company of the reasons are not aligned.